### ISSUE TITLE
[OJS] 3.1.0 tar NOT production ready

### ISSUE THREAD

**Issue by Hutchy68** [pkp/pkp-lib#3103]
I was asked to install OJS for testing purposes and give my opinion of OJS being a viable solution.

Taking the lazy way, I just uploaded the .tar, decompressed to a /ojs/ directory and ran the installer. (**Note** disconcerting to see all the warnings outputting to the screen. Warnings and deprecated, no fatals but still a user shouldn't see them.) 

- Scrolled down and installed.  
- Install went ok 
- Login ok
- Try to create a Journal -  The element is there, but...
```
<a href="myfqdn/ojs/index.php/index/$$$call$$$/grid/admin/journal/journal-grid/create-context"
id="component-grid-admin-journal-journalgrid-createContext-button-5a1667c555de1" title="Create a
New Journal" class="pkp_controllers_linkAction pkp_linkaction_createContext 
kp_linkaction_icon_add_item"></a>
```

`n_add_item"></a>` see the issue? Missing Create Journal between opening and closing `<a>` tags. Mouse pointer is showing the tool tip. Clicking in thin air, nothing happens.

![image](https://user-images.githubusercontent.com/3527469/33160644-0442c2a6-cfec-11e7-8f11-c8afd49d1313.png)

## TL;DR

So I dropped back to 3.0.2 running on php 5.6. It seemed to work fine, up'd to 7.0, things start to break again. Same issue with creating a Journal. Hidden link for Create as above. Dropped back to 5.6 and was able to create a Journal. Testing upgrade to 3.1.0. Upgraded to 3.1.0 from 3.0.2, running php 5.6. Ok, seems to look ok. Cleared caches, Journal menu disappeared, logged out, logged in. Menu back. 

User issue

![image](https://user-images.githubusercontent.com/3527469/33163172-1f08090c-cffb-11e7-8500-2edae7fb54b2.png)

Yep, JS error

```
pkp.min.js:117 Uncaught Error: Constructor for object "$.pkp.controllers.tab.settings.managementSettings.UsersAndRolesTabHandler" not found!
    at Function.a.pkp.classes.Helper.resolveObjectName (pkp.min.js:117)
    at Function.a.pkp.classes.Helper.objectFactory (pkp.min.js:116)
    at HTMLDivElement.<anonymous> (pkp.min.js:443)
    at Function.each (jquery.min.js:2)
    at n.fn.init.each (jquery.min.js:2)
    at n.fn.init.a.fn.pkpHandler (pkp.min.js:443)
    at HTMLDocument.<anonymous> (access:137)
    at j (jquery.min.js:2)
    at Object.fireWith [as resolveWith] (jquery.min.js:2)
    at Function.ready (jquery.min.js:2)
a.pkp.classes.Helper.resolveObjectName @ pkp.min.js:117
a.pkp.classes.Helper.objectFactory @ pkp.min.js:116
(anonymous) @ pkp.min.js:443
each @ jquery.min.js:2
each @ jquery.min.js:2
a.fn.pkpHandler @ pkp.min.js:443
(anonymous) @ access:137
j @ jquery.min.js:2
fireWith @ jquery.min.js:2
ready @ jquery.min.js:2
K @ jquery.min.js:2
```

Turned off CDN to serve files from the install. No change. Hard reload. No change. Click on `Roles`

![image](https://user-images.githubusercontent.com/3527469/33163217-61759ba6-cffb-11e7-84b6-cb98dd2d2c49.png)

Could point out more issues. Error log, just a heck of a lot of Deprecated and Strict Standards. Filling up the log so fast I'd have to either drop their reporting or run a cron job to dump the log.

I haven't tried to install using a git clone. Not sure if I should say try and use the last version of 2.4.8 or just install 3.0.2 for testing. Would really like to see this be a viable solution, but I've have a lot of concerns.

**Comment by asmecher** [comment]
@Hutchy68, OJS 3.1.0 is used in production from the tarball -- I don't think your experience is typical. But rather than debugging it here (where we deal with specific bugs and feature requests), I'd suggest posting at http://forum.pkp.sfu.ca/ where we typically support the community.

**Comment by Hutchy68** [comment]
Hmm, please point me to a production site that’s running 3.1.0. I can’t find one and even your demo is still on 3.0.2.  It is worth a look to see if you’re tarball was packed correctly perhaps some modules weren’t initialize properly before the tar was created. A JS error from cdn or local means something is missing.

**Comment by asmecher** [comment]
@Hutchy68, I'll be happy to help debug, but please post this in the forum to get started.

**Comment by asmecher** [comment]
(Conversation moved to https://forum.pkp.sfu.ca/t/3-1-0-tar-not-production-ready/35634)
